src/Microsoft.Diagnostics.Monitoring.RestServer/HostRestrictionAttribute.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Mvc.ActionConstraints;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.RestServer
+{
+    /// <summary>
+    /// We want to restrict the Prometheus scraping endpoint to only the /metrics call.
+    /// To do this, we determine what port the request is on, and disallow other actions on the prometheus port.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class)]
+    internal sealed class HostRestrictionAttribute : Attribute, IActionConstraintFactory
```

Comment:
I'm worried about that fact that most of this exists in a different form in .NET Core 3.1 and we're replicating it here because we target 2.x.

Reply:
I'm not sure what we can do about that. My other option here would be to not use Attribute routing, but instead use a manually mapped out conventional route with a default route restriction. It will be a bit more tedious to add more routes.

Reply:
Yea that's ok for now. I'll need to play around with this as a library to get a better sense.